bigquery_etl/query_scheduling/telemetry_airflow.py

```diff
@@ -0,0 +1,121 @@
+"""Contains helper methods to interface with the telemetry-airflow repository."""
+
+import attr
+from git import Repo
+import os
+from os.path import isfile, join
+from pathlib import Path
+import re
+import shutil
+import tempfile
+
+
+TELEMETRY_AIRFLOW_GITHUB = "https://github.com/mozilla/telemetry-airflow.git"
+
+
+@attr.s(auto_attribs=True)
+class TelemetryAirflowTaskRef:
+    """Reference to a task defined in telemetry-airflow."""
+
+    dag_name: str = attr.ib()
+    task_name: str = attr.ib()
+    dataset: str = attr.ib()
+    # consists of the table name including version
+    destination_table: str = attr.ib()
+
+
+def download_repository():
+    """Download the telemetry-airflow repository to a temporary directory."""
+    tmp_dir = tempfile.gettempdir() + "/telemetry-airflow/"
+
+    # the repository can only be cloned into an empty directory
+    if os.path.exists(tmp_dir) and os.path.isdir(tmp_dir):
+        shutil.rmtree(tmp_dir)
+
+    Repo.clone_from(TELEMETRY_AIRFLOW_GITHUB, tmp_dir)
+
+    airflow_dag_dir = tmp_dir + "/dags"
+    return airflow_dag_dir
+
+
+def _extract_dag_name(dag_content):
+    """Extract the DAG name for an Airflow DAG definition."""
+    dag_name_regex = [
+        # DAG name inline with DAG definition
+        re.compile(
+            r"DAG\([\n\r\s]*['\"](?P<dag>[^'\"]*)['\"][\n\r\s]*,.*\)", re.DOTALL
+        ),
+        # dag_name defined in variable
+        re.compile(r"dag_name[\n\r\s]*=[\n\r\s]*['\"](?P<dag>[^'\"]*)['\"]", re.DOTALL),
+    ]
+
+    dag_name_matches = [re.findall(regex, dag_content) for regex in dag_name_regex]
+
+    for match in dag_name_matches:
+        if len(match) > 0:
+            return match[0]
+
+    return None
+
+
+def _extract_task_references(dag_content):
+    """Extract destination tables and corresponding tasks from an Airflow DAGs."""
+    dag_name = _extract_dag_name(dag_content)
+
+    airflow_functions_re = (
+        "bigquery_etl_query|bigquery_etl_copy_deduplicate|bigquery_xcom_query"
+    )
+
+    function_calls = re.findall(
+        re.compile(fr"(?:{airflow_functions_re})\((.+?(?=\)\n))\)\n", re.DOTALL),
+        dag_content,
+    )
+
+    task_refs = []
+
+    for fn in function_calls:
+        task_id = re.findall(
+            r"task_id[\n\r\s]*=[\n\r\s]*['\"](?P<task>[^'\"]*)['\"][\n\r\s]*,", fn
+        )
+        table = re.findall(
+            r"destination_table[\n\r\s]*=[\n\r\s]*['\"](?P<tab>[^'\"]*)['\"][\n\r\s]*,",
+            fn,
+        )
+
+        dataset_id = re.findall(
+            r"dataset_id[\n\r\s]*=[\n\r\s]*['\"](?P<dataset>[^'\"]*)['\"][\n\r\s]*,",
+            fn,
+        )
```

Comment:
I decided to use regex to determine dependencies in telemetry-airflow. I looked into using `airflow` directly for parsing DAGs, however there are a lot of additional dependencies that need to be installed, some of the causing conflicts with library versions used here. Plus parsing the DAGs wasn't as fast as just using a regex.

It's probably best to add tests to make sure the regex approach is working correctly.